docs/source/conf.py

```diff
@@ -308,3 +311,26 @@
 
 # How to display URL addresses: 'footnote', 'no', or 'inline'.
 #texinfo_show_urls = 'footnote'
+
+
+try:
+    if Version(__version__).is_prerelease:
+        tags.add("preview")
+
+    is_dirty = bool(subprocess.check_output(["git", "status", "--untracked-files=no", "--porcelain"], text=True).strip())
+    commit_hash = subprocess.check_output(["git", "rev-parse", "HEAD"], text=True).strip()
+    git_describe = subprocess.check_output(["git", "describe", "--tags"], text=True).strip() + ("-dirty" if is_dirty else "")
```

Comment:
I have been very much enjoying the [`versioningit`](https://pypi.org/project/versioningit/) project over the last year or so, which can automate away this sort of stuff (at the expense of reading the version from a `_version.py` file or somesuch instead of using `importlib` metadata). But I tried to write this patch using only what's already present, and it should work fine as long as `git` is present and the docs are being built from the repository, and degrade gracefully if there are problems (i.e. you'll still get a warning box if `git` is not present or the CWD violates this code's assumptions)